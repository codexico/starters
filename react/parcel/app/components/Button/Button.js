import React from 'react';

const Button = (props) => (
  <button
    className="Button"
    type={props.type}
    asdf={props.asdf}
  >
    {props.text}
  </button>
);

export default Button;
