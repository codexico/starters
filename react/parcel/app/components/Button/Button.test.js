import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import Button from './Button';

test('Render Button', () => {
  const button = renderer.create(
    <Button />
  );
  expect(button.toJSON()).toMatchSnapshot();
});

test('Render Button with text', () => {
  const button = renderer.create(
    <Button text={'foo'} />
  );
  expect(button.toJSON()).toMatchSnapshot();
});

test('Render Button with type', () => {
  const button = renderer.create(
    <Button type={'bar'} />
  );
  expect(button.toJSON()).toMatchSnapshot();
});

test('Render Button with asdf', () => {
  const button = renderer.create(
    <Button asdf={'bar'} />
  );
  expect(button.toJSON()).toMatchSnapshot();
});

test('Render Button with foo', () => {
  const button = renderer.create(
    <Button foo={'bar'} />
  );
  expect(button.toJSON()).toMatchSnapshot();
});

test('Render shallow Button with text', () => {
  const button = shallow(
    <Button text={'foo'} />
  );
  expect(button.text()).toBe('foo');
});

test('Render shallow Button with foo', () => {
  const wrapper = mount(
    <Button asdf={'foo'} type={'submit'} />
  );
  expect(wrapper.prop('asdf')).toBe('foo');
});

test('Render shallow Button with bar', () => {
  const wrapper = mount(
    <Button bar={'foo'} />
  );
  expect(wrapper.prop('bar')).toBe('foo');
});
