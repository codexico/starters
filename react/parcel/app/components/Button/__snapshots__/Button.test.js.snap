// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button 1`] = `
<button
  className="Button"
/>
`;

exports[`Render Button with asdf 1`] = `
<button
  asdf="bar"
  className="Button"
/>
`;

exports[`Render Button with foo 1`] = `
<button
  className="Button"
/>
`;

exports[`Render Button with text 1`] = `
<button
  className="Button"
>
  foo
</button>
`;

exports[`Render Button with type 1`] = `
<button
  className="Button"
  type="bar"
/>
`;
